tchannel-core/src/main/java/com/uber/tchannel/messages/ErrorResponse.java

```diff
@@ -30,12 +30,22 @@
     private final ErrorType errorType;
     private final String message;
     private final long id;
+    private final Throwable throwable;
 
     public ErrorResponse(long id, ErrorType errorType, String message) {
         this.id = id;
         this.errorType = errorType;
         this.message = message;
         this.type = FrameType.Error;
+        this.throwable = null;
+    }
+
+    public ErrorResponse(long id, ErrorType errorType, Throwable throwable) {
+        this.id = id;
```

Comment:
you can have a private base constructor, and have both call it with different parameters.